tests/api/controllers/robot_urdf/robot_urdf.c

```diff
@@ -0,0 +1,38 @@
+#include <webots/robot.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "../../../lib/ts_assertion.h"
+#include "../../../lib/ts_utils.h"
+
+#define MAX_RESULT_LENGTH (10 * 1024)
+#define TIME_STEP 16
+
+int main(int argc, char **argv) {
+  ts_setup(argv[0]);
+
+  // Write URDF to file
+  FILE *f;
+  f = fopen("robot.urdf", "w");
+  const char *urdf = wb_robot_get_urdf("");
+  fprintf(f, "%s", urdf);
+  fclose(f);
+
+  // Save output of `check_urdf` to file
+  int urdf_check_status = system("check_urdf robot.urdf > result.txt");
```

Comment:
This is dangerous as it has to be installed separately.
I would rather make this part of the test optional (only if this command is available), and add a second test that compares the result of the export with a know version (committed as an urdf file for example).

Reply:
It supposed to come with ROS installation which is installed for the ROS tests.

I understand, but it is a bit tricky as flexibility compromised. Robot model might change (which we can solve by creating a custom robot model just for this purpose) and making a slight change like changing naming convention or adding more detailed visual elements will cause the test to fail.

Ideally, we should parse the exported URDF and than verify different aspects individually, but that is a lot of work (already done by `check_urdf`).

Reply:
I agree, the solution I proposed will need update each time we change the exporter, but it will at least detect cases where we change it not on purpose.

The check_urdf is indeed the ideal solution, but unfortunately, we want to have the test suite to run on all the platforms whereas ROS is not always installed (we are for example using it on Windows too), so maybe checking it check_urdf is available and use it only in this case is good (it will be for example available on Travis where we run the nightly tests).

Reply:
Good point, I will create one more test then